src/Sarif.Viewer.VisualStudio/ResultTextMarker.cs

```diff
@@ -72,48 +126,169 @@ public Region Region
+        /// <param name="color">The non-highlighted color of the marker.</param>
+        /// <param name="highlightedColor">The highlighted color of the marker.</param>
+        /// <param name="context">The data context for this result marker.</param>
+        public ResultTextMarker(int runIndex, int resultId, Region region, string fullFilePath, string color, string highlightedColor, object context)
+            : this(runIndex: runIndex, resultId: resultId, region: region, fullFilePath: fullFilePath, color: color, highlightedColor: highlightedColor, errorType: null, tooltipContent: null, context: context)
+        {
+        }
 
         /// <summary>
-        /// fullFilePath may be null for global issues.
+        /// Creates a new instances of a <see cref="ResultTextMarker"/>.
         /// </summary>
-        public ResultTextMarker(int runIndex, Region region, string fullFilePath)
+        /// <param name="runIndex">The index of the run as known to <see cref="CodeAnalysisResultManager"/>.</param>
+        /// <param name="resultId">The identifier of the <see cref="SarifErrorListItem"/> that this marker belongs to.</param>
+        /// <param name="region">The original source region from the SARIF log file.</param>
+        /// <param name="fullFilePath">The full file path of the location in the SARIF result.</param>
+        /// <param name="color">The non-highlighted color of the marker.</param>
+        /// <param name="highlightedColor">The highlighted color of the marker.</param>
+        /// <param name="errorType">The error type as defined by <see cref="Microsoft.VisualStudio.Text.Adornments.PredefinedErrorTypeNames"/>.</param>
+        /// <param name="tooltipContent">The tool tip content to display in Visual studio.</param>
+        /// <param name="context">The data context for this result marker.</param>
+        /// <remarks>
+        /// The tool tip content could be as simple as just a string, or something more complex like a WPF/XAML object.
+        /// </remarks>
+        public ResultTextMarker(int runIndex, int resultId, Region region, string fullFilePath, string color, string highlightedColor, string errorType, object tooltipContent, object context)
         {
-            this.runIndex = runIndex;
+            this.ResultID = resultId;
+            this.RunIndex = runIndex;
+            this.FullFilePath = fullFilePath;
             this.region = region ?? throw new ArgumentNullException(nameof(region));
-            FullFilePath = fullFilePath;
-            Color = DEFAULT_SELECTION_COLOR;
+            this.Color = color;
+            this.HighlightedColor = highlightedColor;
+            this.ToolTipeContent = tooltipContent;
+            this.ErrorType = errorType;
+            this.Context = context;
+        }
+
+        public IEnumerable<ISarifLocationTag> GetTags<T>(ITextBuffer textBuffer, IPersistentSpanFactory persistentSpanFactory)
+            where T : ITag
```

Comment:
Comment this, especially the fact that only our tagger implementation calls this.